packages/@lwc/engine-core/src/renderer.ts

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+export type HostNode = any;
+export type HostElement = any;
+
+type N = HostNode;
+type E = HostElement;
+
+//
+// Primitives
+//
+
+export let ssr: boolean;
+export function setSsr(ssrImpl: boolean) {
```

Comment:
What do you think about creating a single setter accepting an object instead of multiple setters? 

The only downside I see with this approach is that it will increase the bundle size a little, but it should preserve the same performance benefits.

Reply:
I'll need to test to see what the impact is. I suspect you're right that the minified size will not be as small, but the runtime perf would be about the same.